### PR TITLE
Fixing the recommendation of using INIT_SOLR_HOME for new versions

### PR DESCRIPTION
**Pull request by jebnix**
https://issues.apache.org/jira/browse/SOLR-XXXXX

# Description

The docs says:

> Solr docker supports a INIT_SOLR_HOME setting, which copies the contents

but [this section](https://github.com/apache/solr/blob/main/solr/solr-ref-guide/modules/deployment-guide/pages/solr-in-docker.adoc#updating-from-solr-5-7-to-8) says:

> The custom SOLR_HOME can no longer be used because various scripts depend on the new locations. Consequently, INIT_SOLR_HOME is also no longer supported.

So that seems like an invalid suggestion.

# Solution

Removing this part/ saying it is valid only until version 7.

# Tests

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `main` branch.
- [x] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [x] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

**Comment by janhoy** [comment]
Thanks for giving the docs some love. Lots of the docker related docs is slightly outdated, and was copied from the old docker-solr repo into the ref guide.

Apart from `INIT_SOLR_HOME` no longer being supported (I found no reference to it), Solr will also now  accept an empty `SOLR_HOME` and will automatically fall back to defaults for `solr.xml` and `zoo.cfg`. So there is no need for special init anymore. But any custom folder you point to must already exist.

**Comment by jebnix** [comment]
@janhoy [The reference for `INIT_SOLR_HOME` not being supported](https://github.com/apache/solr/blob/main/solr/solr-ref-guide/modules/deployment-guide/pages/solr-in-docker.adoc#updating-from-solr-5-7-to-8) says:


> The custom SOLR_HOME can no longer be used because various scripts depend on the new locations. Consequently, INIT_SOLR_HOME is also no longer supported.

File: solr/solr-ref-guide/modules/deployment-guide/pages/docker-faq.adoc [hunk 1]
```diff
@@ -140,7 +140,7 @@ docker run -it -v $PWD/mysolrhome:/mysolrhome -e SOLR_HOME=/mysolrhome solr
 
 This does need a pre-configured directory at that location.
 
-To make this easier, Solr docker supports a INIT_SOLR_HOME setting, which copies the contents
+Until version 7, Solr docker supported an INIT_SOLR_HOME setting, which copies the contents
```
Comment (janhoy) [inline solr/solr-ref-guide/modules/deployment-guide/pages/docker-faq.adoc:143]: Better, just remove the entire mention of `INIT_SOLR_HOME`. We don't normally document earlier versions in ref guide

**Comment by epugh** [comment]
@jebnix  this one too I think could be improved by just removing the updates..  Heck, this docs are going on Solr 10 and Solr     9 :-)

**Comment by janhoy** [comment]
Closing this, since #2277 will rewrite the entire paragraph.

Also, Solr can now work with an empty `SOLR_HOME` so this was old info anyway.

**Pull request closed without merging**